desktop/ui/src/main/java/org/datacleaner/panels/fuse/StreamColumnMatrixMultipleCoalesceUnitPropertyWidget.java

```diff
@@ -120,19 +119,34 @@ private void refresh(final CoalesceUnit[] units) {
         // build registries of available columns and coalesced columns
         final Multimap<Table, InputColumn<?>> allTablesAndColumns = ArrayListMultimap.create();
         final Multimap<Table, InputColumn<?>> coalescedTablesAndColumns = ArrayListMultimap.create();
-        {
-            if (units != null) {
-                for (CoalesceUnit unit : units) {
-                    final InputColumn<?>[] coalescedInputColumns = unit.updateInputColumns(inputColumns).getInputColumns();
-                    for (InputColumn<?> inputColumn : coalescedInputColumns) {
-                        final Table table = sourceColumnFinder.findOriginatingTable(inputColumn);
-                        coalescedTablesAndColumns.put(table, inputColumn);
+
+        for (InputColumn<?> inputColumn : inputColumns) {
```

Comment:
Yes, but we've never really done foreach `final`s and method signature `final`s. I'm all for starting to do that, but then it needs to be a decision we make overall.

I'll skip this for now, then we can discuss it in next retro.